### PR TITLE
Feature label names

### PR DESCRIPTION
**Pull request by harryli0088**
This PR adds the ability for users of equine to save the natural language names of the classes/labels (ex `["Ankle boot", "Shirt", ...]`) as well as features for tabular data (ex `["duration","fiat_mean",...]`) that the model was trained on.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -43,3 +48,45 @@ def random_dataset(draw):
...
+    return new_model, tmp_filename
+
+# return a list of random strings
+# based off https://stackoverflow.com/a/34485032
+def generate_random_string_list(list_length:int, str_length:int=3):
+    chars = ascii_lowercase + digits
+    return [''.join(choice(chars) for _ in range(str_length)) for _ in range(list_length)]
+
+
+def load_equine_model(model_path):
```
Comment (harryli0088) [inline tests/conftest.py:82]: Move this into equine utils? Check if it already exists?

File: tests/conftest.py [hunk 2]
```diff
@@ -43,3 +48,45 @@ def random_dataset(draw):
...
     return dataset, num_classes, way
+
+
+def use_basic_embedding_model(random_dataset):
+    dataset, num_classes, _ = random_dataset
+    X, _ = dataset.tensors
+    embedding_model = BasicEmbeddingModel(X.shape[1], num_classes)
+    return dataset, num_classes, X, embedding_model
+
+def use_save_load_model_tests(model, X, tmp_filename:str="tmp_eq.pt"):
```
Comment (harryli0088) [inline tests/conftest.py:59]: Make sure the tmp file names are different to avoid race conditions if tests are run in parrallel

File: tests/test_equine_gp.py [hunk 3]
```diff
@@ -154,3 +120,47 @@ def test_equine_gp_save_load_with_vis(random_dataset) -> None:
...
+    # without feature and label names
+    model = eq.EquineGP(embedding_model, num_classes, num_classes)
+    loss_fn = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(
+        model.parameters(),
+        lr=0.001,
+        momentum=0.9,
+        weight_decay=0.0001,
+    )
+    model.train_model(dataset, loss_fn, optimizer, 10)
```
Comment (harryli0088) [inline tests/test_equine_gp.py:139]: Change the number epochs to like 2 or something small since we don't care about training here, and label it for clarity

File: tests/test_equine_gp.py [hunk 4]
```diff
@@ -154,3 +120,47 @@ def test_equine_gp_save_load_with_vis(random_dataset) -> None:
...
+
+    model = eq.EquineGP(embedding_model, num_classes, num_classes, feature_names=feature_names,label_names=label_names)
+    loss_fn = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(
+        model.parameters(),
+        lr=0.001,
+        momentum=0.9,
+        weight_decay=0.0001,
+    )
+    model.train_model(dataset, loss_fn, optimizer, 10)
```
Comment (harryli0088) [inline tests/test_equine_gp.py:158]: Change the number epochs to like 2 or something small since we don't care about training here, and label it for clarity

File: tests/test_equine_protonet.py [hunk 5]
```diff
@@ -225,46 +224,57 @@ def test_predict_fail_before_training(random_dataset):
...
+
+    assert ( new_model.get_feature_names() is None ), "feature_names changed on reload"
+    assert ( new_model.get_label_names() is None ), "label_names changed on reload"
+
+    # with feature and label names
+    feature_names = generate_random_string_list(X.shape[1])
+    label_names = generate_random_string_list(num_classes)
+
+    model = eq.EquineProtonet(embedding_model, num_classes, feature_names=feature_names,label_names=label_names)
+    model.train_model(dataset, num_episodes=10)
```
Comment (harryli0088) [inline tests/test_equine_protonet.py:271]: Change the number epochs to like 2 or something small since we don't care about training here

File: tests/test_equine_protonet.py [hunk 6]
```diff
@@ -225,46 +224,57 @@ def test_predict_fail_before_training(random_dataset):
...
+        os.remove(tmp_filename)  # Cleanup
+
+
+@given(random_dataset=random_dataset())
+@settings(deadline=None, max_examples=1)
+def test_equine_protonet_save_load_with_feature_and_label_names(random_dataset) -> None:
+    dataset, num_classes, X, embedding_model = use_basic_embedding_model(random_dataset)
+
+    # without feature and label names
+    model = eq.EquineProtonet(embedding_model, num_classes)
```
Comment (harryli0088) [inline tests/test_equine_protonet.py:258]: Change the number epochs to like 2 or something small since we don't care about training here

**Review by nukularrr — APPROVED** [review]
We performed this review together; it is approved.
